change_environment.py

```diff
@@ -6,78 +6,94 @@
-        ends = view.find_all(end_re, sublime.IGNORECASE)
-        # compile the begin_re (findall does not work if its compiled)
-        begin_re = re.compile(begin_re)
-
-        comment_line_re = re.compile(r"\s*%.*")
-
-        def is_comment(reg):
-            line_str = view.substr(view.line(reg))
-            return comment_line_re.match(line_str) is not None
-        begins = [b for b in begins if not is_comment(b)]
-        ends = [e for e in ends if not is_comment(e)]
-
-        def extract_begin_region(region):
-            """creates a sublime.Region: \\begin{|text|}"""
-            s = view.substr(region)
-            boffset = len("\\begin{")
-            m = begin_re.search(s)
-            if m:
-                boffset = m.regs[1][0]
-            return sublime.Region(region.begin() + boffset, region.end() - 1)
-
-        def extract_end_region(region):
-            """creates a sublime.Region: \\end{|text|}"""
-            boffset = len("\\end{")
-            return sublime.Region(region.begin() + boffset, region.end() - 1)
-
-        new_regions = []
-        one_sel = len(view.sel()) == 1
-
-        for sel in view.sel():
-            # partition the open and closed environments
-            begin_before, begin_after =\
-                _partition(begins, lambda b: b.begin() <= sel.begin())
-            end_before, end_after =\
-                _partition(ends, lambda e: e.end() < sel.begin())
-
-            # get the nearest open environments
-            try:
-                begin = _get_closest_begin(begin_before, end_before)
-                end = _get_closest_end(end_after, begin_after)
-            except NoEnvError as e:
-                if one_sel:
-                    sublime.status_message(e.args[0])
-                    return
-                else:
-                    continue
-
-            # extract the regions for the environments
-            begin_region = extract_begin_region(begin)
-            end_region = extract_end_region(end)
-
-            # validity check: matching env name
-            if view.substr(begin_region) == view.substr(end_region):
-                new_regions.append(begin_region)
-                new_regions.append(end_region)
-            elif one_sel:
-                sublime.status_message(
-                    "The environment begin and end does not match:"
-                    "'{0}' and '{1}'"
-                    .format(view.substr(begin_region), view.substr(end_region))
-                )
-        if not new_regions:
-            sublime.status_message("Environment detection failed")
-            return
-
+        new_regions = _find_env_regions(view)
         view.sel().clear()
         for r in new_regions:
             view.sel().add(r)
 
+class LatexToggleEnvironmentStarCommand(sublime_plugin.TextCommand):
+    def run(self, edit):
+        view = self.view
+        new_regions = _find_env_regions(view)
+        # 
+        if view.substr(new_regions[0])[-1] == '*':
```

Comment:
Can you please change this to `if view.substr(new_regions[0]).endswith('*'):` so it won't throw an error for empty strings. In addition you should add a safety check and return if `new_regions` is empty.